CMakeLists.txt

```diff
@@ -1,114 +1,144 @@
-# top level CMakeLists.txt
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
-SET_PROPERTY(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
-
-set(CMAKE_CXX_STANDARD 17)
-add_compile_options(-pedantic)
-add_compile_options(-pedantic-errors)
-add_compile_options(-Wextra)
-add_compile_options(-Wunused-variable)
-add_compile_options(-Wdisabled-optimization)
-add_compile_options(-Wall)
-
-
-# Disabled by default, used to be necessary on the Cedar cluster of Compute Canada
-#add_compile_options(-fext-numeric-literals)
-
-#add_compile_options(-Werror)
-#add_compile_options(-Weffc++)
-
-option(WARNINGERROR "Consider compilation warnings as errors" OFF)
-IF(WARNINGERROR)
-    add_compile_options(-Werror)
-ENDIF()
-
-FIND_PACKAGE(deal.II 9.4.0 QUIET
-  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
-  )
-IF(NOT ${deal.II_FOUND})
-  MESSAGE(FATAL_ERROR "\n"
-    "*** Could not locate a (sufficiently recent) version of deal.II. ***\n\n"
-    "You may want to either pass a flag -DDEAL_II_DIR=/path/to/deal.II to cmake\n"
-    "or set an environment variable \"DEAL_II_DIR\" that contains this path."
-    )
-ENDIF()
-
-DEAL_II_INITIALIZE_CACHED_VARIABLES()
-
-IF(EXISTS ${CMAKE_SOURCE_DIR}/bundled/CMakeLists.txt)
-  SET(DEAL_II_HAVE_BUNDLED_DIRECTORY TRUE)
-ENDIF()
-
-IF(EXISTS ${CMAKE_SOURCE_DIR}/doc/CMakeLists.txt)
-  SET(DEAL_II_HAVE_DOC_DIRECTORY TRUE)
-ENDIF()
-
-IF(EXISTS ${CMAKE_SOURCE_DIR}/tests/CMakeLists.txt)
-  SET(DEAL_II_HAVE_TESTS_DIRECTORY TRUE)
-ENDIF()
-
-#set(PROTOTYPES "FALSE" CACHE BOOL)
-option(PROTOTYPES "Enable prototype applications" OFF)
-option(MARCH_NATIVE "Enable the -march=native flag" OFF)
-
-
-Project(lethe)
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-ENABLE_TESTING()
-ADD_SUBDIRECTORY(source)
-ADD_SUBDIRECTORY(applications)
-ADD_SUBDIRECTORY(applications_tests)
-ADD_SUBDIRECTORY(tests)
-IF(PROTOTYPES)
-    MESSAGE("Adding prototypes")
-    ADD_SUBDIRECTORY(prototypes)
- ENDIF()
- IF(MARCH_NATIVE)
-    MESSAGE("  Enabling -march=native flag. This may cause some tests to break.")
+cmake_minimum_required(VERSION 3.8...3.21)
+if(CMAKE_VERSION VERSION_LESS 3.12) # To support the version range
+  cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+endif()
+
+# TARGET_SUPPORTS_SHARED_LIBS is set by PROJECT, but we have to call
+# FIND_PACKAGE before that, and deal.II's configuration calls
+# ADD_LIBRARY with the SHARED keyword, so setting
+# TARGET_SUPPORTS_SHARED_LIBS beforehand may be necessary in some
+# environments.
+set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
+
+# Assume that deal.II_DIR won't be provided on the command line in favor
+# of DEAL_II_DIR and can thus be used to detect whether CMake was
+# already run.
+if(DEFINED deal.II_DIR)
+  set(_had_deal_ii TRUE)
+endif()
```

Comment:
This was added in commit https://github.com/lethe-cfd/lethe/pull/563/commits/d53052892fd59a3e014a44bebaffc018246047f4, the message of which explains the rationale: it's not CMake that is prevented from rerunning (because of course that would be quite incorrect), rather only the status messages related to deal.II features are prevented from printing, because when you rerun CMake on the same build directory, you most often will have already come to terms with the deal.II features which are available.

Reply:
I see. Sorry, I did not look at the individual commit message. They are kinda hard to get to from the github interface which I use to review the PR.